ui/app/hooks/useTokenTracker.js

```diff
@@ -56,6 +65,10 @@ export function useTokenTracker (tokens) {
     return teardownTracker
   }, [teardownTracker])
 
+  const previousTokens = usePrevious(tokens)
+  const previousUserAddress = usePrevious(userAddress)
```

Comment:
you can pass isEqual as a second argument to useSelector to have it do deep comparison and return the same value if inputs are deeply equal... though we've only had to do this in the case where the values are serialized from another value meaning they are always different objects... if there is referential equality then useSelector doesn't return a new object.

tl;dr - I don't think you need to use the usePrevious hook for the values you're pulling from redux .. if you do then just pass isEqual as a second arg to useSelector.